sympy/printing/codeprinter.py

```diff
@@ -69,6 +69,7 @@ class CodePrinter(StrPrinter):
     # may be supported
     # function_to_rewrite : (function_to_rewrite_to, iterable_with_other_functions_required)
     _rewriteable_functions = {
+            'cot': ('tan', []),
```

Comment:
Presumably this should also be done for sec, csc, coth, sech, csch as well.

Reply:
And all the inverses asec, acsc etc

Reply:
Good observation! I'll add those as well (but maybe not tests for all combinations etc...).

Reply:
I think it is worth testing all combinations. Some of them won't work and some of the rewrite rules seem to be wrong or broken e.g.:
```python
In [2]: sech(x).rewrite(cosh)
Out[2]: sech(x)

In [3]: sech(x).rewrite(sinh)
Out[3]: 
   1   
───────
cosh(x)

In [4]: csch(x).rewrite(cosh)
Out[4]: 
   1   
───────
sinh(x)

In [5]: csch(x).rewrite(sinh)
Out[5]: csch(x)
```
Also if any rewrite rule is missing then `lambdify` goes into infinite recursion so that would happen for `asech` here:
```python
In [9]: asec(x).rewrite(acos)
Out[9]: 
    ⎛1⎞
acos⎜─⎟
    ⎝x⎠

In [10]: asech(x).rewrite(acosh)
Out[10]: asech(x)
```

Reply:
OK! I'll update the current one then. I found that the `a*h` could not be rewritten, but not the others.

Reply:
Some are simply not supported, while others are getting rewritten as part of automatic evaluation. For example:
```
In [3]: sech(x).rewrite(sinh)
Out[3]: 
   1   
───────
cosh(x)
```

Reply:
There should be a `sech._eval_rewrite_as_cosh` method.

Reply:
Oh, I see you've added it now